api/data/NMBS/liveboard.php

```diff
@@ -139,18 +138,34 @@ private static function parseData($xml, $time, $lang, $fast = false)
                 $delay = $d[1] * 3600 + $d[2] * 60;
             }
 
+            $platform = '?'; // Indicate to end user platform is unknown
+            $platformNormal = true;
+            if (isset($journey['platform'])) {
+                $platform = (string) $journey['platform'];
+            }
+            
+            if (isset($journey['newpl'])) {
+                $platform = (string) $journey['newpl'];
+                $platformNormal = false;
+            }
+            
+            $time = '00d'.(string) $journey['fpTime'].':00';
+            preg_match("/(..)\/(..)\/(..)/si", (string) $journey['fpDate'], $dayxplode);
+            $dateparam = '20'.$dayxplode[3].$dayxplode[2].$dayxplode[1];
+
+            $unixtime = tools::transformtime($time, $dateparam);
+
             if ($fast) {
                 $stationNode = new Station();
-                $stationNode->name = (string) $journey['dir'];
+                $stationNode->name = (string) $journey['targetLoc'];
```

Comment:
`targetLoc` is always there and has always the same value as `dir`, but the latter is missing if requesting arrivals. 

> On 29 Dec 2015, at 12:20, Pieter Colpaert <notifications@github.com> wrote:
> 
> In api/data/NMBS/liveboard.php:
> 
> >              if ($fast) {
> >                  $stationNode = new Station();
> > -                $stationNode->name = (string) $journey['dir'];
> > +                $stationNode->name = (string) $journey['targetLoc'];
> Why this change? Should we also include 'dir' as another entity called direction? Might be more interesting than targetLoc in some situations?
> 
> —
> Reply to this email directly or view it on GitHub.
>